### PR TITLE
Code quality/increase coverage of firmware management service

### PR DESCRIPTION
**Pull request by andrevanduin**

### CODE REVIEW

File: osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java [hunk 1]
```diff
@@ -144,7 +144,7 @@ private void createSsldPendingFirmwareUpdateRecord(final CorrelationIds ids, fin
             LOGGER.info("Saved pending fimware update record for SSLD: {}, {}", ids.getDeviceIdentification(),
                     ssldPendingFirmwareUpdate);
         } catch (final Exception e) {
-            LOGGER.error("Caugth exception when creating pending firmware update record for SSLD: {}",
+            LOGGER.error("Caught exception when creating pending firmware update record for SSLD: {}",
```
Comment (arjen0203) [inline osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java:147]: INSANE FIX

**Review by arjen0203 — COMMENTED** [review]
looks good

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementServiceTest.java [hunk 2]
```diff
@@ -401,4 +441,222 @@ void checkSsldPendingFirmwareUpdateReturnsTrueAndDeletesPendingUpdateWithMatchin
...
+        final CorrelationIds ids = this.getCorrelationIds();
+        final List<FirmwareVersionDto> versionsOnDevice = new ArrayList<>();
+
+        this.firmwareManagementService.handleGetFirmwareVersionResponse(versionsOnDevice, ids, "messageType", 1,
+            ResponseMessageResultType.OK, this.defaultException);
+		
+        verify(this.webServiceResponseMessageSender).send(this.responseMessageCaptor.capture());
+		verify(this.ssldPendingFirmwareUpdateRepository, never()).delete(any());
+
+		final ResponseMessage responseMessage = this.responseMessageCaptor.getValue();
```
Comment (HighCrit) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementServiceTest.java:613]: Hele gekke indentations jonge man

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementServiceTest.java [hunk 3]
```diff
@@ -401,4 +441,222 @@ void checkSsldPendingFirmwareUpdateReturnsTrueAndDeletesPendingUpdateWithMatchin
...
+    	final SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = Mockito.mock(SsldPendingFirmwareUpdate.class);
+    	final List<SsldPendingFirmwareUpdate> ssldPendingFirmwareUpdates = Arrays.asList(ssldPendingFirmwareUpdate);
+   
+    	when(this.ssldPendingFirmwareUpdateRepository.findByDeviceIdentification(any()))
+    		.thenReturn(ssldPendingFirmwareUpdates);
+    	when(ssldPendingFirmwareUpdate.getCorrelationUid()).thenReturn("differentUid");
+
+        this.firmwareManagementService.handleGetFirmwareVersionResponse(versionsOnDevice, ids, "messageType", 1,
+    		ResponseMessageResultType.OK, null);
+
```
Comment (HighCrit) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementServiceTest.java:650]: Hele gekke indentations jonge man

**Review by HighCrit — CHANGES_REQUESTED** [review]

**Review by HighCrit — APPROVED** [review]

**Pull request merged**